### PR TITLE
Need generate the code to get base type property if the derived override property is internal

### LINKED ISSUE #21716: Need use base type property get method when the override property is internal
When serialize the following derived type, it will throw “access method DerivedTypeWithDifferentOverrides.get_Name4() failed”. This is because the generated code is wrong and it try to access the internal Name4 property.

```
public class BaseType
{
    public virtual string Name1 { get; set; }
    public string Name2 { get; set; }
    public string Name3 { get; set; }
    public string Name4 { get; set; }
    public string @Name5 { get; set; }
}

public class DerivedTypeWithDifferentOverrides : BaseType
{
    public override string Name1 { get; set; }
    new public string Name2 { get; set; }
    new public string Name3 { get; set; }
    new internal string Name4 { get; set; }
    new public string Name5 { get; set; }
}
```

### PR DESCRIPTION
**Pull request by huanwu**
@shmao @zhenlan @mconnew 
Fix #21716

### CODE REVIEW

**Comment by shmao** [comment]
We need to skip the test on netfx since desktop also has the issue.

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 1]
```diff
@@ -5273,4 +5273,24 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
...
+{
+    public virtual string Name1 { get; set; }
+    public string Name2 { get; set; }
+    public string Name3 { get; set; }
+    public string Name4 { get; set; }
+    public string @Name5 { get; set; }
+}
+
+
+
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: Remove  extra empty lines.

File: src/System.Private.Xml/src/System/Xml/Serialization/Types.cs [hunk 2]
```diff
@@ -1194,6 +1194,14 @@ private static bool ShouldBeReplaced(MemberInfo memberInfoToBeReplaced, Type der
                             replacedInfo = info;
                             if (replacedInfo != memberInfoToBeReplaced)
                             {
+                                if(memberInfoToBeReplaced is PropertyInfo propertyInfoToBeReplaced)
+                                {
+                                    if (!info.GetMethod.IsPublic && propertyInfoToBeReplaced.GetMethod.IsPublic)
```
Comment (shmao) [inline src/System.Private.Xml/src/System/Xml/Serialization/Types.cs]: We probably can check `!info.GetMethod.IsPublic` first before checking `memberInfoToBeReplaced is PropertyInfo propertyInfoToBeReplaced`. In most cases, `info.GetMethod.IsPublic` would be true and thus we can avoid the other checks.

**Comment by shmao** [comment]
The test is failing in ReflectionOnly mode.

```
Message :
System.InvalidOperationException : There was an error generating the XML document.\r\n---- System.InvalidOperationException : Internal error: Could not find member named Name4 of type DerivedTypeWithDifferentOverrides.
Stack Trace :
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle, String id)
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle)
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces)
   at System.Xml.Serialization.XmlSerializer.Serialize(Stream stream, Object o, XmlSerializerNamespaces namespaces)
   at System.Xml.Serialization.XmlSerializer.Serialize(Stream stream, Object o)
   at XmlSerializerTests.SerializeAndDeserialize[T](T value, String baseline, Func`1 serializerFactory, Boolean skipStringCompare, XmlSerializerNamespaces xns)
   at XmlSerializerTests.DerivedTypeWithDifferentOverrides()
----- Inner Stack Trace -----
   at System.Xml.Serialization.ReflectionXmlSerializationHelper.GetMember(Type declaringType, String memberName)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.GetMemberValue(Object o, String memberName)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteStructMethod(StructMapping mapping, String n, String ns, Object o, Boolean isNullable, Boolean needType)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteElement(Object o, ElementAccessor element, Boolean writeAccessor)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteElements(Object o, Object enumSource, ElementAccessor[] elements, TextAccessor text, ChoiceIdentifierAccessor choice, Boolean writeAccessors, Boolean isNullable)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteMember(Object o, Object choiceSource, ElementAccessor[] elements, TextAccessor text, ChoiceIdentifierAccessor choice, TypeDesc memberTypeDesc, Boolean writeAccessors)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.GenerateTypeElement(Object o, XmlTypeMapping xmlMapping)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteObjectOfTypeElement(Object o, XmlTypeMapping mapping)
   at System.Xml.Serialization.ReflectionXmlSerializationWriter.WriteObject(Object o)
   at System.Xml.Serialization.XmlSerializer.SerializeUsingReflection(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle, String id)
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle, String id)
```

**Comment by huanwu** [comment]
File issue #21731 to track the test failure in reflectiononly mode.

**Comment by huanwu** [comment]
@dotnet-bot test this please

**Pull request merged**